sdk/scala-sdk-testkit/src/main/scala/com/akkaserverless/scalasdk/testkit/impl/EventSourcedEntityEffectsRunner.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.scalasdk.testkit.impl
+
+import com.akkaserverless.scalasdk.eventsourcedentity.EventSourcedEntity
+import com.akkaserverless.scalasdk.testkit.EventSourcedResult
+import com.akkaserverless.scalasdk.testkit.impl.EventSourcedResultImpl
+import scala.collection.immutable.Seq
+
+import com.akkaserverless.scalasdk.eventsourcedentity.CommandContext
+
+abstract class EventSourcedEntityEffectsRunner[S](entity: EventSourcedEntity[S]) {
+  var _state: S
+  var events: Seq[Any]
+  val commandContext: CommandContext
```

Comment:
Ah, missed this in the review, they shouldn't be public, could perhaps be protected instead, but better if they can stay private and just move the getters up to this class and pass them as parameters to where they need to be assigned etc.

Reply:
Well spotted 👍 
Like the other access modifiers it would need to be `protected`. `abstract member may not have private modifier`.

Reply:
I think it would be better with the field made non abstract and private and pull the access methods up here instead of generating them.

Reply:
totally. I followed your suggestions. Updated now.